### PR TITLE
Assigning positions on memory reader is persistent

### LINKED ISSUE #2018: Inconsistent assignation behavior of the memory reader
**Expected behavior**

Any operation that modify the coordinates on a time step is permanent when using the memory reader. Switching frames do not erase the changes.

**Actual behavior**

Applied to `u.trajectory.ts.positions`, operations that **modify in place** the underlying array are kept permanent (_e.g._ `+=`) while operations that replaces the array are lost when switching frames.

Everything works as expected when working on an AtomGroup.

The behavior is thus not consistent. It also breaks @davidercruz's `rotateby` transformation when applied to the memory reader (see #2000).

**Code to reproduce the behavior**

Show us how to reproduce the failiure. If you can, use trajectory files from the test data.

``` python
import numpy as np
import MDAnalysis as mda

n_frames = 5
n_atoms = 4
n_dims = 3

total_coords = n_frames * n_atoms * n_dims
shape = (n_frames, n_atoms, n_dims)
coords = np.arange(total_coords, dtype=np.float32).reshape(shape)

u = mda.Universe(coords)
print(u.atoms.positions)

new_positions = u.atoms.positions + 10
print(new_positions)

## Replace the coordinates
u.trajectory.ts.positions = new_positions
#u.atoms.positions = new_positions
print(u.trajectory.ts.positions)
print(u.atoms.positions)  # these are the new coordinates

u.trajectory[0]
print(u.trajectory.ts.positions)  # the new coordinates are lost

## Modify the coordinates
u.trajectory.ts.positions *= 4
print(u.trajectory.ts.positions)  # these are the new coordinates

u.trajectory[0]
print(u.trajectory.ts.positions)  # these are still the new coordinates
```

### PR DESCRIPTION
**Pull request by jbarnoud**
Fixes #2018

Assigning an array to ts.positions with the memory reader does not break
the link to trajectory.coordinate_array anymore. That link needs to be
reset when switching to a new frame. This is done with the new method
ts.replace_position_array.

PR Checklist
------------
 - [X] Tests?
 - [X] Docs?
 - [X] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

**Comment by tylerjereddy** [comment]
Windows test failures have increased by more than 70 -- probably because of a previously merged PR; I suggest coming up with a solution to prevent this as I will otherwise have to compete with all changes from all devs that break windows stuff (i.e., I will probably fail).

**Comment by orbeckst** [comment]
> Windows test failures have increased by more than 70 

Can you identify the PR, e.g., using `git bisect`? I know that's additional work.

I think the problem is that on all PRs the Appveyor result is essentially ignored as it just presents the green check mark. I am not sure what the best way forward is. Perhaps decorate the remaining 20 or so failing tests with XFAIL on windows and then require Appveyor to pass, too. Perhaps we should discuss on the dev list? @tylerjereddy could you take the lead on this discussion, please?

File: package/MDAnalysis/coordinates/memory.py [hunk 1]
```diff
@@ -196,20 +196,25 @@ class Timestep(base.Timestep):
...
+    array.
     """
-
-    @property
-    def positions(self):
-        return base.Timestep.positions.fget(self)
-
-    @positions.setter
-    def positions(self, new):
+    def replace_positions_array(self, new):
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/memory.py]: It sounds as if this should not really be used by the average user. Perhaps make it underscore-named? `_replace_positions_array()`?

File: package/MDAnalysis/coordinates/memory.py [hunk 2]
```diff
@@ -196,20 +196,25 @@ class Timestep(base.Timestep):
...
+        """Replace the array of positions
+
+        Replaces the array of positions by an other array. This is different
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/memory.py]: another

File: package/MDAnalysis/coordinates/memory.py [hunk 3]
```diff
@@ -196,20 +196,25 @@ class Timestep(base.Timestep):
...
+        from the bahavior of the :attr:`position` property that replaces the
+        **content** of the array. The :meth:`replace_positions_array` method
+        should only be used to set the positions to a different frame; there,
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/memory.py]: in :meth:`MemoryReader._read_next_timestep`

File: package/MDAnalysis/coordinates/memory.py [hunk 4]
```diff
@@ -196,20 +196,25 @@ class Timestep(base.Timestep):
...
+        the memory reader sets the positions to a view to the correct frame.
+        Modifying the positions for a given frames should be done with the
+        :attr:`positions` attribute that do not break the link between the
+        array of positions in the time step and the
+        :attr:`MemoryReader.coordinate_array`.
+        """
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/memory.py]:
```rest
... versionchanged: 0.19.0
```

**Comment by jbarnoud** [comment]
I solved it earlier today and Tyler merged it. I do not remember the pr number, though. Le 5 août 2018 7:39 PM, Oliver Beckstein <notifications@github.com> a écrit :
Windows test failures have increased by more than 70

Can you identify the PR, e.g., using git bisect? I know that's additional work.
I think the problem is that on all PRs the Appveyor result is essentially ignored as it just presents the green check mark. I am not sure what the best way forward is. Perhaps decorate the remaining 20 or so failing tests with XFAIL on windows and then require Appveyor to pass, too. Perhaps we should discuss on the dev list? @tylerjereddy could you take the lead on this discussion, please?

**Comment by orbeckst** [comment]
If I don't get to it in the next few hours (after Travis), please feel free to _merge_.

**Comment by orbeckst** [comment]
Rebased against develop. I (or someone else) should do a merge commit when Travis has run (again).

**Comment by orbeckst** [comment]
@zemanj I fixed a few things that I found, if you have anything else just push to the branch. Feel free to merge when you're happy. (Should be a merge commit and not a rebase so that (1) @jbarnoud 's authorship stays intact and (2) the benchmarks are run on merge commits.)

**Comment by jbarnoud** [comment]
Thanks @zemanj for the merge and welcome as a core dev 👍

**Pull request merged**